src/components/SchedulePanel/components/OfficeHoursDialog/index.js

```diff
@@ -0,0 +1,526 @@
+
+    // Require that content havint appropriate size and format
+    switch (fieldName) {
+      case 'descriptionInput':
+        descriptionValid = value.trim() !== '' && value.length < 50;
+        fieldValidationErrors.descriptionInput = descriptionValid ? '' : 'Description is not Valid';
+        break;
+      case 'locationInput':
+        locationValid = value.trim() !== '' && value.length < 50;
+        fieldValidationErrors.locationInput = locationValid ? '' : 'Location is not Valid';
+        break;
+      case 'startHourInput':
+        startHourValid =
+          value.trim() !== '' &&
+          parseInt(this.state.endHourInput, 10) > parseInt(value, 10) &&
+          parseInt(value, 10) >= 5 &&
+          parseInt(value, 10) <= 10;
+        fieldValidationErrors.startHourInput = startHourValid ? '' : 'Start Time is not Valid';
+        break;
+      case 'endHourInput':
+        endHourValid =
+          value.trim() !== '' &&
+          parseInt(this.state.startHourInput, 10) < parseInt(value, 10) &&
+          parseInt(value, 10) >= 5 &&
+          parseInt(value, 10) <= 22;
+        fieldValidationErrors.endHourInput = endHourValid ? '' : 'End Time is not Valid';
+        break;
+      default:
+        break;
+    }
+    this.setState(
+      {
+        formErrors: fieldValidationErrors,
+        descriptionValid: descriptionValid,
+        locationValid: locationValid,
+        startHourValid: startHourValid,
+        endHourValid: endHourValid,
+      },
+      () => {
+        this.validateForm();
+      },
+    );
+  }
+
+  validateForm() {
+    this.setState({
+      formValid:
+        this.state.locationValid &&
+        this.state.descriptionValid &&
+        this.state.startHourValid &&
+        this.state.endHourValid &&
+        this.state.checkedValid,
+    });
+    console.log("Form Valid ", this.state.formValid);
+  }
+
+  handleSubmit = e => {
+    e.preventDefault();
+
+    var startHour = this.state.startHourInput;
+    var endHour = this.state.endHourInput;
+    var location = this.state.locationInput;
+    var description = this.state.descriptionInput;
+
+    var mySchedule = {
+      startHour: startHour,
+      endHour: endHour,
+      location: location,
+      description: description,
+      monday: this.state.checkedDayofWeek.checkedMo,
+      tuesday: this.state.checkedDayofWeek.checkedTu,
+      wednesday: this.state.checkedDayofWeek.checkedWe,
+      thursday: this.state.checkedDayofWeek.checkedTh,
+      friday: this.state.checkedDayofWeek.checkedFr,
+      saturday: this.state.checkedDayofWeek.checkedSa,
+      sunday: this.state.checkedDayofWeek.checkedSu,
+      allDay: this.state.checkedC,
+    };
+    this.props.onDialogSubmit(mySchedule);
+
+    this.handleClose();
+  };
+
+  validateCheck = dayofWeek => {
+    var valid = false;
+    for (let day in dayofWeek) {
+      if (dayofWeek[day]) {
+        valid = true;
+      }
+    }
+    this.setState({ checkedValid: valid }, () => {
```

Comment:
Once again, is there a reason that the following function has to be run within setState?